src/OmniSharp.Roslyn.CSharp/Helpers/LocationExtensions.cs

```diff
@@ -17,22 +19,55 @@ public static QuickFix GetQuickFix(this Location location, OmniSharpWorkspace wo
             var lineSpan = Path.GetExtension(location.SourceTree.FilePath).Equals(".cake", StringComparison.OrdinalIgnoreCase)
                 ? location.GetLineSpan()
                 : location.GetMappedLineSpan();
-            var path = lineSpan.Path;
-            var documents = workspace.GetDocuments(path);
 
-            var line = lineSpan.StartLinePosition.Line;
-            var text = location.SourceTree.GetText().Lines[line].ToString();
+            var documents = workspace.GetDocuments(lineSpan.Path);
+            var sourceText = GetSourceText(location, documents, lineSpan.HasMappedPath);
+            var text = GetText(location, sourceText, lineSpan.StartLinePosition.Line);
 
             return new QuickFix
             {
-                Text = text.Trim(),
-                FileName = path,
-                Line = line,
+                Text = text,
```

Comment:
Curious: Should this be the entire text encompassed by the line mapping or just the first line? If it's the latter it might make sense to change the `GetText` method to be `GetLineText` to make it more clear

Reply:
it should be the line, I changed it accordingly 👍